verification/curator-service/api/test/model/automation.test.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    AutomationDocument,
+    automationParsingValidator,
+    automationSchema,
+} from '../../src/model/automation';
+
+import { Error } from 'mongoose';
+import fullModel from './data/automation.full.json';
+import regexParsingModel from './data/regex-parsing.full.json';
+import mongoose from 'mongoose';
+
+const Automation = mongoose.model<AutomationDocument>(
+    'Automation',
+    automationSchema,
+);
+
+// Used to test automationParsingValidator.
+const wrapperSchema = new mongoose.Schema({
+    automation: {
+        type: automationSchema,
+        validate: automationParsingValidator,
+    },
+});
+type WrapperDocument = mongoose.Document & { automation: AutomationDocument };
+const Wrapper = mongoose.model<WrapperDocument>('Wrapper', wrapperSchema);
+
+describe('validate', () => {
+    it('an automation without a schedule is invalid', async () => {
+        const missingSchedule = { ...fullModel };
+        delete missingSchedule.schedule;
+
+        return new Automation(missingSchedule).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
+    });
+
+    it('an automation without either a parser or regexParsing is invalid', async () => {
+        const noParsing = { ...fullModel };
+        delete noParsing.parser;
+        const wrapper = { automation: noParsing };
+
+        return new Wrapper(wrapper).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
+    });
+
+    it('an automation with both a parser and regexParsing is invalid', async () => {
```

Comment:
maybe one for w/o either a parser or regexParsing?

Reply:
Sorry, that's the one above. This isn't completely clear, since the loaded model contains only a parser (which it must, because parser/regexParsing are mutually exclusive).